### PR TITLE
Swift: Add edge from last node of variable inline to boolean literal cleanup

### PR DESCRIPTION
**Pull request by Adarsh-NP**
This PR adds edge from last node of variable inline clusters (replace_identifier_with_value, replace_self_identifier_with_value) to boolean literal cleanup to accommodate recursive cleanups. 

This fixes the cases where after variable inline we would have the need to invoke the boolean literal cleanups. Please refer to tests for more info.

This PR also includes adding (init_declaration) to enclosing_node to achieve variable inline for variables declared inside init of a class through delete_field_initialisation. (Additional fix over #471)

Refer to doc: [Variable Inline](https://docs.google.com/document/d/1iVyocClx9gK095ruvoWrjTtn5w6aVAyTee-CskkrCEw/edit#)

### CODE REVIEW

**Review by ketkarameya — CHANGES_REQUESTED** [review]
Can u add a test scenario for this ?

File: src/cleanup_rules/swift/edges.toml [hunk 1]
```diff
@@ -13,6 +13,20 @@
 # The edges in this file specify the flow between the rules.
 
 
+[[edges]]
+scope = "Parent"
+from = "replace_expression_with_boolean_literal"
+to = ["boolean_literal_cleanup", "variable_inline_cleanup"]
```
Comment (Adarsh-NP) [inline src/cleanup_rules/swift/edges.toml:19]: Here I've just changed the order of rules.

File: src/cleanup_rules/swift/edges.toml [hunk 2]
```diff
@@ -38,22 +52,22 @@ to = [
 
 [[edges]]
 scope = "Parent"
-from = "replace_expression_with_boolean_literal"
-to = ["boolean_literal_cleanup", "variable_inline_cleanup"]
+from = "replace_identifier_with_value"
```
Comment (ketkarameya) [inline src/cleanup_rules/swift/edges.toml:55]: is this change overlapping with the other PR?

File: src/cleanup_rules/swift/rules.toml [hunk 3]
```diff
@@ -814,8 +819,11 @@ not_contains = [
     )""",
 ]
 
+# also run the rule for the variables declared in init
+# [[rules.filters]]
+# enclosing_node = "(init_declaration) @cid"
 
-# rule to delete field initialised in the init subject to the mentioned filters
+# rule to delete field declared in class but initialised in the init subject to the mentioned filters
```
Comment (ketkarameya) [inline src/cleanup_rules/swift/rules.toml:826]:
```suggestion
# rule to delete field declared in a class and initialized in the init,  subject to the mentioned filters
```

File: src/cleanup_rules/swift/rules.toml [hunk 4]
```diff
@@ -785,7 +790,7 @@ replace = ""
 is_seed_rule = false
 
 [[rules.filters]]
-enclosing_node = "(class_declaration) @cd"
+enclosing_node = "[(class_declaration) (init_declaration)] @cd"
```
Comment (ketkarameya) [inline src/cleanup_rules/swift/rules.toml:793]: Is this change related to this PR?

File: test-resources/swift/variable_inline/field_variable_inline/input/FieldVariableInline.swift [hunk 5]
```diff
@@ -10,10 +10,9 @@
 // limitations under the License.
```
Comment (ketkarameya) [inline test-resources/swift/variable_inline/field_variable_inline/input/FieldVariableInline.swift:10]: please format this file before, with swift formatter.

File: test-resources/swift/variable_inline/local_variable_inline/expected/VariableInlining.swift [hunk 6]
```diff
@@ -60,14 +60,8 @@ class C7 {
     }
```
Comment (ketkarameya) [inline test-resources/swift/variable_inline/local_variable_inline/expected/VariableInlining.swift:60]: please format this file with swift formatter

File: test-resources/swift/variable_inline/local_variable_inline/input/VariableInlining.swift [hunk 7]
```diff
@@ -78,16 +78,13 @@ class C7 {
     }
 }
```
Comment (ketkarameya) [inline test-resources/swift/variable_inline/local_variable_inline/input/VariableInlining.swift:79]: same as above

**Review by ketkarameya — COMMENTED** [review]
left comments (none of the comments are nits)

File: src/cleanup_rules/swift/rules.toml (hunk 4 above)
Comment (Adarsh-NP) [reply src/cleanup_rules/swift/rules.toml:793]: This change is not directly related to the PR but solves the issue of inlining variables declared in init scope in the class itself. Earlier fix provided in #471 was not enough. This separates the flow and ensures things work correctly. Since this was just a one line change, I just included here.

File: src/cleanup_rules/swift/edges.toml (hunk 2 above)
Comment (Adarsh-NP) [reply src/cleanup_rules/swift/edges.toml:55]: Yeah, this is because nature of newly added tests are such. Landing either of PRs would solve it for all.

File: test-resources/swift/variable_inline/local_variable_inline/input/VariableInlining.swift (hunk 7 above)
Comment (Adarsh-NP) [reply test-resources/swift/variable_inline/local_variable_inline/input/VariableInlining.swift:79]: Done

File: src/cleanup_rules/swift/edges.toml (hunk 2 above)
Comment (ketkarameya) [reply src/cleanup_rules/swift/edges.toml:55]: hmmmm. Ok. But u will have to rebase the other one,

File: src/cleanup_rules/swift/rules.toml (hunk 4 above)
Comment (ketkarameya) [reply src/cleanup_rules/swift/rules.toml:793]: update the PR description for the PR, and mark this issue as related (And close it once u land the PR)

**Review by ketkarameya — APPROVED** [review]
nits, otherwise LGTM

**Review by ketkarameya — APPROVED** [review]
nits, otherwise LGTM
